backend/src/v5/middleware/permissions/components/teamspaces.js

```diff
@@ -53,4 +53,15 @@ TeamspacePerms.isTeamspaceMember = async (req, res, next) => {
 	}
 };
 
+TeamspacePerms.hasMemberAccessToTeamspace = async (req, res, next) => {
+	const { params } = req;
+	const { teamspace, member } = params;
+	const hasAccess = await hasAccessToTeamspace(teamspace, member);
+	if (hasAccess) {
+		next();
+	} else {
+		respond(req, res, createResponseCode(templates.notAuthorized, 'Member does not have access to the teamspace'));
```

Comment:
It should really be a 404.

It's not that the user didn't have sufficient access to do the request, it's that the member it's trying to request doesn't exist.